app/src/main/java/com/example/countrylist/base/views/BaseView.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.countrylist.base.views
+
+interface BaseView<T> {
+    fun setPresenter(presenter: T)
+    fun hideProgressBar()
+    fun showProgressBar()
+    fun showError()
```

Comment:
We can customise the error by passing string parameter:
 fun showError(message: String)